### DISCUSSION TITLE
Is it possible to get values of some of the attributes of an element at webview?

### DISCUSSION THREAD

**Discussion by proximilian** [leancodepl/patrol#1071] [Q&A]
Hi Guys, 

My use case is the following:
In a webview I have the `resource-id` of a web element that I can use as a `Selector` to interact with it thanks to the great native feature of Patrol. But, however, I want to get the `text` attribute that it has. The text is not visible on the screen. In my particular case the web element is a QR code picture which `text` attribute I want to get as a String. Is that possible? And in general, can we get any attribute of a web element by having a selector?
Here is a screenshot from the inspector in order to be more clear:
![image](https://user-images.githubusercontent.com/125548662/224114332-3d77e8c0-5062-4441-8606-036b1314b536.png)

**Comment by bartekpacia** [comment]
Hi @proximilian – thanks for creating this Discussion.

This is not possible currently, but it seems like we should have it.

Pinging @mateuszwojtczak and @jBorkowska (other members of Patrol core team) to voice their opinions on this and how they think about it.

**Reply by proximilian** [reply]
Thanks guys,

Our QA team will be really really grateful because, unfortunately, that use case is blocking us in some way in our project. We are searching for a workaround but so far without any success. 
One more related(this time not native) use case comes to my mind: User to be able to get the visual text of a widget e.g. user can find a widget by a `Key`, but may wants to get the text of it as a `String` if it(the text) is somehow different at each test run.

Thanks in advance for you work and dedication to Patrol. It's awesome!

**Reply by jBorkowska** [reply]
Hi @proximilian, thank you for giving Patrol a try :)

Actually there is such a thing - you can use `.text` on a Patrol finder to get text of a widget (more info [here](https://pub.dev/documentation/patrol/latest/patrol/PatrolFinder/text.html)). Check it out if you want.

**Reply by proximilian** [reply]
Nice to hear that. Although it is not our current use case, I'll try it out of curiosity. I may need it in some future cases. Thank you for the info.
We are not only just giving Patrol a try. Our expectations and hopes are towards it. :) Congrats on your work.

**Reply by bartekpacia** [reply]
> Actually there is such a thing - you can use .text on a Patrol finder to get text of a widget (more info [here](https://pub.dev/documentation/patrol/latest/patrol/PatrolFinder/text.html)). Check it out if you want.

Unfortunately it works only for Flutter widgets – not for native views :(

**Reply by bartekpacia** [reply]
Thinking about it now, you can try doing:

```dart
final views = await $.native.getNativeViews(Selector(resourceId: "resource-id"));
expect(views.length, equals(1)); // I understand that only one view with such resourceId exists
final view = views.first;
// now you can access properties of `view`
```

**Reply by proximilian** [reply]
> > Actually there is such a thing - you can use .text on a Patrol finder to get text of a widget (more info [here](https://pub.dev/documentation/patrol/latest/patrol/PatrolFinder/text.html)). Check it out if you want.
> 
> Unfortunately it works only for Flutter widgets – not for native views :(

yes, yes. I gоt it. But it was useful to know it as I can use it in other cases in future. Now our use case is with native views.

**Reply by proximilian** [reply]
> Thinking about it now, you can try doing:
> 
> ```dart
> final views = await $.native.getNativeViews(Selector(resourceId: "resource-id"));
> expect(views.length, equals(1)); // I understand that only one view with such resourceId exists
> final view = views.first;
> // now you can access properties of `view`
> ```

To be honest in the beginning I was exactly wondering about that `getNativeViews` method. I'll try your suggestion as a workaround. Thank you.

**Reply by bartekpacia** [reply]
Let me know if this worked.

**Reply by proximilian** [reply]
I tried it on a button in a webwiew, but unfortunately it doesn't work. It seems  that the `getNativeViews` method is returning no elements at the list for me since the test is failing at the `expect` statement -> ( Expected: <1> Actual: <0>). However in a new test run I tried to tap on the button using the same `contentDesctription` Selector as with the `getNativeViews` and it is tapped successfully. So the element can be found without problem. Btw I tried with  `contentDesctription`  (not with resourceId)  but I think it doesn't matter at all.

**Comment by jBorkowska** [comment]
I like this idea, I'll add this to our backlog.

**Reply by proximilian** [reply]
Do I need to create a new feature-request issue or you will add it directly to the backlog?

**Reply by jBorkowska** [reply]
I will add it directly :)

**Comment by bartekpacia** [comment]
For the record, this is tracked in #1087
